tests/pamiq_core/trainer/test_container.py

```diff
@@ -0,0 +1,58 @@
+from typing import override
+
+import pytest
+from pytest_mock import MockerFixture
+
+from pamiq_core.data import DataUser, DataUsersDict
+from pamiq_core.model import InferenceModel, TrainingModel, TrainingModelsDict
+from pamiq_core.state_persistence import PersistentStateMixin
+from pamiq_core.trainer import Trainer, TrainersDict
+
+
+class DummyTrainer(Trainer):
+    @override
+    def train(self) -> None:
+        pass
+
+
+class TestTrainersDict:
+    def test_trainers_dict_subclasses(self):
+        assert issubclass(TrainersDict, PersistentStateMixin)
+
+    @pytest.fixture
+    def trainer(self, mocker: MockerFixture) -> Trainer:
+        trainer = DummyTrainer()
+        trainer.attach_training_models_dict(
+            training_models_dict=mocker.Mock(TrainingModelsDict)
+        )
+        return trainer
```

Comment:
Trainerはテストの主体ではないのでモック
```suggestion
    @pytest.fixture
    def mock_trainer(self, mocker: MockerFixture) -> Trainer:
        mock = mocker.Mock(Trainer)
        mock.is_trainable.return_value = True
        return mock
```

さらに`is_trainable`が`False`のMockもあるといいかも

```py
    @pytest.fixture
    def mock_trainer_not_trainable(self, mocker: MockerFixture) -> Trainer:
        mock = mocker.Mock(Trainer)
        mock.is_trainable.return_value = False
        return mock
```